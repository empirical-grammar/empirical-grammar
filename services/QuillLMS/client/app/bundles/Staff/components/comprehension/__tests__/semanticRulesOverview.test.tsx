import * as React from 'react';
import { shallow } from 'enzyme';

import SemanticRulesOverview from '../semanticRules/semanticRulesOverview';
import LabelsTable from '../semanticRules/labelsTable';
import ModelsTable from '../semanticRules/modelsTable';

const mockProps = {
  activityId: '17',
  prompts: [{ id: 1 }, { id: 2 }, { id: 3 }]
};

describe('SemanticRulesOverview component', () => {
  const container = shallow(<SemanticRulesOverview {...mockProps} />);

  it('should render SemanticRulesOverview', () => {
    expect(container).toMatchSnapshot();
  });

  it('should render a Labels Table and Models table for each prompt', () => {
    expect(container.find(LabelsTable).length).toEqual(3);
    expect(container.find(ModelsTable).length).toEqual(3);
  });
});
