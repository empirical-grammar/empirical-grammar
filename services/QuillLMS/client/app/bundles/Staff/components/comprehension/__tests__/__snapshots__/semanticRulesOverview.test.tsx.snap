// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SemanticRulesOverview component should render SemanticRulesOverview 1`] = `
<div
  className="semantic-rules-overview-container"
>
  <section
    className="prompt-section"
    key="1"
  >
    <LabelsTable
      activityId="17"
      prompt={
        Object {
          "id": 1,
        }
      }
    />
    <ModelsTable
      activityId="17"
      prompt={
        Object {
          "id": 1,
        }
      }
    />
  </section>
  <section
    className="prompt-section"
    key="2"
  >
    <LabelsTable
      activityId="17"
      prompt={
        Object {
          "id": 2,
        }
      }
    />
    <ModelsTable
      activityId="17"
      prompt={
        Object {
          "id": 2,
        }
      }
    />
  </section>
  <section
    className="prompt-section"
    key="3"
  >
    <LabelsTable
      activityId="17"
      prompt={
        Object {
          "id": 3,
        }
      }
    />
    <ModelsTable
      activityId="17"
      prompt={
        Object {
          "id": 3,
        }
      }
    />
  </section>
</div>
`;
